lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/client/AzureBatchRuntimeConfigurationCreator.java

```diff
@@ -47,27 +44,39 @@
    * Get or create a {@link ConfigurationModule} for the Azure Batch runtime.
    *
    * @param isWindows true if Azure Batch pool nodes run Windows, false otherwise.
+   * @param isDockerContainer true if Azure Batch pool nodes are container-based, false otherwise.
    * @return the configuration module object.
    */
-  public static ConfigurationModule getOrCreateAzureBatchRuntimeConfiguration(final boolean isWindows) {
+  public static ConfigurationModule getOrCreateAzureBatchRuntimeConfiguration(
+      final boolean isWindows,
+      final boolean isDockerContainer) {
 
     if (AzureBatchRuntimeConfigurationCreator.conf == null) {
       ConfigurationModuleBuilder builder = AzureBatchRuntimeConfigurationStatic.CONF;
-      ConfigurationModule module;
+
       if (isWindows) {
-        module = builder.bindImplementation(CommandBuilder.class, WindowsCommandBuilder.class).build();
+        builder = builder.bindImplementation(CommandBuilder.class, WindowsCommandBuilder.class);
       } else {
-        module = builder.bindImplementation(CommandBuilder.class, LinuxCommandBuilder.class).build();
+        builder = builder.bindImplementation(CommandBuilder.class, LinuxCommandBuilder.class);
+      }
+
+      if (isDockerContainer) {
+        builder = builder.bindImplementation(TcpPortProvider.class, ListTcpPortProvider.class);
       }
```

Comment:
Should this really be hard-coded? What if someone creates a different Provider type?

Reply:
I don't see examples where the avro file can bind to an implementation. It only provides parameter bindings. Won't fix for now. If you disagree, we can discuss about this.

---
In reply to: [197881586](https://github.com/sharathmalladi/reef/pull/5#discussion_r197881586) [](ancestors = 197881586)